### ISSUE TITLE
Cinnamon (2.8 & 3.0) break using amdgpu-pro with RX460

### ISSUE THREAD

**Issue by JollyWizard** [linuxmint/Cinnamon#5698]
I recently bought an AMD Radeon RX460.  It's a slightly newer card, and it's only supported on linux in the `amdgpu-pro` package (plain `amdgpu` does not recognize the card, including the `ppa:oibaf` version).  Without, I am stuck in default ati/VESA/software mode when using Linux, which eats into my system responsiveness.

I have tried the official BETA driver package (`amdgpu-pro`) on multiple fresh installs to mixed results.  I also tried updating from `ppa:oibaf/graphics-drivers` before installing the drivers, which actually broke a working Ubuntu system.

Results of my os testing:

```
Windows 10 :: catalyst :: Works fine. OS & Borderlands 2 (Card is good) 
Ubuntu 16.04 :: Works fine. OS & Borderlands 2 (Mint should work?)
Ubuntu 16:04 :: w/ `ppa:oibaf/...` first :: Broken desktop. (see below)
Mint 17.3 :: Broken desktop.  (Compilation issue) (see below)
Mint 18 :: Broken desktop.  (see below)
Mint 18 :: w/ `ppa:oibaf/...` first :: Broken desktop. (see below)
```

The drivers are also supposed to work for Ubuntu 14.04, but I haven't tested them.

I'd like to hunt down the reason why it doesn't work on cinnamon, but it does on Ubuntu.
## Compilation issue

On Mint 17.3 (Cinnamon 2.8?) The drivers install, but there is a build error during the DKMS stage.  Rather than being mostly black, the background and taskbar render incompletely (see symptoms summary below).  Windows will appear, but their contents are incomplete and the system lags/hangs when trying to switch between windows and focus elements.
## Broken Desktop Symptoms

I think it's important to mention here, that in all of my broken desktop results, the login screen works fine and all of it's elements are rendered.  It's only after the transition to the desktop environment that everything breaks.  I assume this is because the login screen runs in software mode.

The desktop is usually black, and usually with a single white pixel in the upper left corner.  The mouse is rendered and moves.

Sometimes, desktop elements, such as the icons or the taskbar, are rendered, but they are generally incomplete. The icons don't render their text.  Icons are not rendered on the taskbar.  Using keyboard shortcuts, terminal windows can be opened, and you can see the mouse change to a clock when over a thinking window, but the window is invisible.

If the service (either `lightdm` or `mdm`) is restarted, more or less of the desktop elements may be rendered.  A complete system restart usually leads to less elements being rendered.
## Provided diagnostiscs

I didn't collect diagnostic information during my initial testing, but I installed the driver package, collected some files, and attached them here, before uninstalling again.

I used Mint 18 for the diagnostics and was able to capture a screenshot of the single pixel using the invisible terminal with the `import` command.  I also saved the Xorg.0.log to a file, and ran `inxi`.  I also provided for comparison, the inxi output on the working system, after I ran `amdgpu-pro-uninstall`
## Theories and diagnosis?

I'm not clear why it works with Ubuntu 16.04 and not Mint 18, but hopefully that means its simple to diagnose and fix. Please advise on information to provide and I will set up and test.
# Attached Files

[ming18.amdgpu-pro.inxi.txt](https://github.com/linuxmint/Cinnamon/files/478319/ming18.amdgpu-pro.inxi.txt)
[mint18.ati.inxi.txt](https://github.com/linuxmint/Cinnamon/files/478320/mint18.ati.inxi.txt)
[Xorg.0.log.mint18.amdgpu-pro.txt](https://github.com/linuxmint/Cinnamon/files/478321/Xorg.0.log.mint18.amdgpu-pro.txt)
[mint18 amdgpu-pro screenshot](https://cloud.githubusercontent.com/assets/4997950/18610293/3ab9cee2-7ce6-11e6-9dcd-7cf1699910e1.png)

**Comment by leigh123linux** [comment]
File you issue against the PPA that supplied the amdgpu-pro as this isn't cinnamon related.

**Comment by atomi** [comment]
By the way the RX cards were only added to amdgpu in kernel 4.7  and you'll need the polaris10 and polaris11 firmware files in `/lib/firmware/amdgpu` when installing the kernel, but the performance sucks and it'ss buggy (read crashes on Mint 18, cinnamon anyway). The amdgpu-pro drivers are even worse. AMD should have pushed hard on getting the RX line backported to 4.4 imo. As it is now, I'm returning the card because I refuse to use Windows.

**Comment by double-facepalm** [comment]
Could you please give this some priority? It seems to affect all Radeon RX cards. AMDGPU-PRO is the official recommended driver so it should be supported by Cinnamon.

Please let me know if you require any help with diagnostics / testing. At the moment, I'm running Mint 18. Cinnamon ends up in a black screen straight after login. I've installed MATE as a workaround and that seems to work perfectly fine, including all my Steam games.

Thanks in advance,

O.

**Comment by leigh123linux** [comment]
I don't see why you think this is a cinnamon issue , I bet the AMDGPU-PRO isn't good enough to run gnome-shell either (mate runs fine with any 2D driver, cinnamon requires a 3D  driver)!
The issue is clearly a broken graphics driver so please file a report against it as it's unlikely that the issue will be resolved here.

**Comment by double-facepalm** [comment]
Didn't realise MATE runs in 2D mode. Anyway, all my games run fine with the driver, including DOTA2 and Talos using both OpenGL and Vulkan. That's a pretty good achievement on AMD card to be honest.

Is there any troubleshooting we can do? Can the Cinnamon team replicate the issue?

**Comment by double-facepalm** [comment]
Looks like it's been a driver issue indeed. AMD have just released AMDGPU-PRO 16.40. I can now log in back to Cinnamon again! There is still some issues with the driver though (I think they admit it in the official release notes). In my case I can notice micro lags (stuttering) in benchmarks like Unigine Heaven. It doesn't affect the results but it seems the quality of the output and the smoothness is much worse compared to the same benchmark ran under MATE. I'll open that in a separate Issue.

**Comment by axdoceo** [comment]
Hi, If you are usign the Linux Mint 18 with kernel 4.4,  the solution is update the kernel from 4.4 to  4.8, in the Udate Manager of Mint you can update the kernel easily, after the reboot you can install the last drivers AMDGPU-PRO 17.10 without problems.
